proto/tendermint/abci/types.proto

```diff
@@ -424,6 +424,10 @@ message ExecTxResult {
 // TxResult contains results of executing the transaction.
 //
 // One usage is indexing transaction results.
+// TODO(thane): Remove this struct. It's effectively only used in
+//              types/events.go#EventDataTx and these fields could easily be
+//              introduced directly into that struct. There doesn't seem to be
+//              any other use for this struct via any other interface.
```

Comment:
is there a reason to not do this, are there issues to track the todos?

Reply:
Not sure there's an issue for this. I have it in one of my "should be done yesterday" checklists, tho :-)
The point of this is that `TxResult` is defined in ABCI but not used here, and hence we needed to name `ExecTxResult` like that, rathen than `TxResult` to avoid name conflicts.
So if we could move `TxResult` definition away from abci protos (or, as @thanethomson suggests, get rid of it), then we could rename `ExecTxResult` to `TxResult` in the abci interface.

Reply:
#8463